aya/src/maps/mod.rs

```diff
@@ -300,14 +293,7 @@ impl Map {
             return Err(MapError::AlreadyCreated { name: name.into() });
         }
         let map_path = path.as_ref().join(name);
-        let path_string = match CString::new(map_path.to_str().unwrap()) {
-            Ok(s) => s,
-            Err(e) => {
-                return Err(MapError::InvalidPinPath {
-                    error: e.to_string(),
-                })
-            }
-        };
+        let path_string = CString::new(map_path.to_str().unwrap()).unwrap();
```

Comment:
The only possible error here is `NulError`. That would require the map name to end with a nul byte - which if it ever did have one was (hopefully) lost when the `object` lib converted the symbol name into a Rust string.